### DISCUSSION TITLE
Data may be lost during transfer in eBpf

### DISCUSSION THREAD

**Discussion by WhatDaniel** [gojue/ecapture#291] [Q&A] [answered]
HI, i know ebpf a little, and i used to make some test about ebpf. If there are much more data in kernel space, and the receive process in user space is slower than ebpf, so some data which need to be transferred to user space tend to be lost. I  wanna know how to solve the case.

**Comment by SeeFlowerX** [comment]
same question 😢

**Comment by cfc4n** [comment, marked as answer]
1. Filter and reduce the amount of data written from kernel space. 
2. Increase the buffer size of ebpf map.  like `rd, err := perf.NewReader(em, os.Getpagesize()*64)` via https://github.com/gojue/ecapture/blob/bf9dee6f4b41abfcea44f27a4c0df6a2134e373c/user/module/imodule.go#L193
3. Speed up the consumption rate in user space.
